### ISSUE TITLE
Tabsets aren't tabbing as expected

### ISSUE THREAD

**Issue by nathanj3** [quarto-dev/quarto-cli#10284]
### Bug description

Tabsets don't seem to be tabbable in the current version of Quarto (which is a problem for accessibility). I found an [older issue](https://github.com/quarto-dev/quarto-cli/issues/730) that mentioned some inconsistency but stated that the [tabs in the tabset documentation](https://quarto.org/docs/interactive/layout.html#tabset-panel) were working as expected. However, when I check the tabset documentation, it has the same issue as in my minimal example: Tab navigation seems to ignore the tabsets and their content entirely. 

### Steps to reproduce

````
# Title

::: {.panel-tabset}

## First tab

Text goes here

## Second tab

Other text might go here

:::
````

### Expected behavior

Tabsets should be tab navigable, which is a [feature in Bootstrap](https://getbootstrap.com/docs/5.0/components/navs-tabs/#working-with-flex-utilities) (here's another [example](http://web-accessibility.carnegiemuseums.org/code/tabs/#apples) from an external site),  

### Actual behavior

Tab navigation skips the tabset.

### Your environment

- IDE: RStudio 2024.04.2 Build 764, "Chocolate Cosmos" Release (e4392fc9, 2024-06-05) for windows
- R version 4.4.1 (2024-06-14 ucrt) -- "Race for Your Life"
- Windows 10
- I tried using both Firefox 127.0.1 (32-bit) and Chrome (Version 126.0.6478.127) and the issue was present in both browsers.

### Quarto check output

````
Microsoft Windows [Version 10.0.19045.4529]
(c) Microsoft Corporation. All rights reserved.

Z:\Qualtrics\QualtricsBook>quarto check
Quarto 1.5.53
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.53
      Path: C:\Users\[redacted]\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: Installation From Path
      Path: C:\texlive\2023\bin\windows
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.4.1
      Path: C:/PROGRA~1/R/R-44~1.1
      LibPaths:
        - C:/Users/[redacted]/AppData/Local/R/win-library/4.4
        - C:/Program Files/R/R-4.4.1/library
      knitr: 1.46
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK
````

**Comment by cscheid** [comment]
> ### Bug description
> Tabsets don't seem to be tabbable in the current version of Quarto (which is a problem for accessibility). I found an [older issue](https://github.com/quarto-dev/quarto-cli/issues/730) that mentioned some inconsistency but stated that the [tabs in the tabset documentation](https://quarto.org/docs/interactive/layout.html#tabset-panel) were working as expected. 

This is admittedly confusing because of the "tabset" name.

But, just to clarify, our docs don't say anything about "tabset navigation" in the sense of hitting the "tab" key.  I agree our "tab-key navigation" needs improvement, but our documentation doesn't currently promise that.

**Comment by glin** [comment]
+1 here, and I would consider it an accessibility bug in that Tabset Panels aren't usable by keyboards or screen readers at all right now. The fix could be fairly straightforward from a quick look at the tabset html though. 

The Tabset tabs use anchor tags which need an `href` attribute to be accessible. If I manually add an empty `href=""`, the tabs work as expected - you can tab-navigate to the tabset and then switch tabs using arrow keys. There already seems to be accessible markup and JavaScript for making the tabsets keyboard and screen reader usable. The `href` could optionally be an actual `#anchor` link to the tabbed content.
```diff
<a
+ href=""
  class="nav-link active"
  id="tabset-1-1-tab"
  data-bs-toggle="tab"
  data-bs-target="#tabset-1-1"
  role="tab"
  aria-controls="tabset-1-1"
  aria-selected="true"
  >Plot</a>
```

Alternatively, as the [Bootstrap docs recommend](https://getbootstrap.com/docs/5.2/components/navs-tabs/#accessibility), a `<button>` could work better than an anchor, as it'll automatically have keyboard support, e.g.:
```diff
-<a
+<button
  class="nav-link active"
  id="tabset-1-1-tab"
  data-bs-toggle="tab"
  data-bs-target="#tabset-1-1"
  role="tab"
  aria-controls="tabset-1-1"
  aria-selected="true"
-  >Plot</a>
+  >Plot</button>
```

For expected behavior, here's another example I found in the Bootstrap docs: https://getbootstrap.com/docs/5.2/components/navs-tabs/#javascript-behavior
- Tabset Panels should be navigable when pressing the Tab key. The first tab should be visibly focused
- Once a tab is focused, other tabs should be switchable using arrow keys (this already seems to work, but I couldn't verify it when manually editing the tabs to use `<button>`s)
- Should be able to navigate out of the Tabset Panel from any tab using the Tab key (this already seems to work)

**Comment by srvanderplas** [comment]
This is an issue that I'm concerned with because it causes most of my content to fail WCAG2.1 checks. Quarto should, by default, produce documents which are at least minimally compliant with accessibility standards wherever possible.

**Comment by cscheid** [comment]
@srvanderplas Thanks for the feedback. Can you share what tool you're using to produce the error, and the specific error being reported? It's hard for us to do something about the problem without any information besides "fails the check".

(As a minor point, I think "are [not] at least minimally compliant" is an unfair description of Quarto's a11y behavior.)

One problem I can see is that in our `quarto preview` functionality, we are using `tabster`, [Microsoft's own open-source tool for keyboard navigation in html tabs](https://github.com/microsoft/tabster/). And tabster's authors specifically claim that one of the warnings produced by the tools [is a false positive](https://github.com/microsoft/tabster/issues/288).

As a result, in our own internal checks, we are following Microsoft's guidance and removing tabster's dummy elements. See [`src/resources/formats/html/axe/axe-check.js`](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/axe/axe-check.js).

We really do want to help with a11y, and we really do work on it. But we also really do need requests that provide info we can act on!

**Comment by cscheid** [comment]
To expand further -

If I don't use `quarto preview` for navigating the pages (instead calling `quarto publish` or `quarto render` and a local server), then even if I remove that tabster exclusion from our axe-core check, I get no accessibility violation reports.

(The upshot is that I really do need to know which specific tool is being used to check for violations, and unfortunately we're blocked on helping until we know more.)

**Comment by cscheid** [comment]
Finally, I seem to be able to tab to the tabs, at least using `main` and the provided document:

## .qmd

````qmd
---
title: wcag tab check
format:
  html:
    axe:
      output: document
---

::: {.panel-tabset}

## First tab

Text goes here

## Second tab

Other text might go here

:::
````

## Demo

This is a video recording of me using the tab key to get to the tabset headers, then keyboard arrows to navigate specific tabs (which I believe to be the expected behavior):

https://github.com/user-attachments/assets/2b24421e-8ddd-4473-a946-4f9ad3e04ddb

**Comment by glin** [comment]
I tested this again with Quarto 1.7.32 and still see the tabbing issue, but it seems like it is specific to Quarto websites. With @cscheid's html example above, tabbing works perfectly fine, and those tabs have the empty `href` attributes.

To repro, the Quarto website still has the issue: https://quarto.org/docs/interactive/layout.html#tabset-panel
Click on the Tabset Panel heading anchor, then press Tab, and focus skips straight to the next code block after the tabs.

Or also, I was testing this locally like:
```sh
$ quarto --version
1.7.32

$ quarto create project website mysite
``` 

And then for `index.qmd`:
```qmd
---
title: "mysite"
---

This is a Quarto website.

To learn more about Quarto websites visit <https://quarto.org/docs/websites>.


::: {.panel-tabset}

## First tab

Text goes here

## Second tab

Other text might go here

:::
```
Unlike basic html documents, the tabsets in website projects don't have the empty `href` attributes.

**Comment by mcanouil** [comment]
@glin please try the latest pre-release or the development version of Quarto CLI, which are closer to what Carlos has tested.

**Comment by glin** [comment]
@mcanouil Done, I tested with the latest prerelease and see the same results. Tabs work fine in html documents, but not in website projects.
```sh
$ quarto --version
1.8.14
```

**Comment by glin** [comment]
To be more specific, it only affects websites using a Bootswatch theme (default, cosmo, etc). I just found https://github.com/quarto-dev/quarto-cli/issues/730#issuecomment-1112322315 and confirmed that websites with `theme: none` have tabbing work fine, in 1.8.14.

**Comment by srvanderplas** [comment]
I was doing a manual audit - I don't recall this being caught by the IBM
accessibility checker browser addon I have been using.

**Comment by cscheid** [comment]
> I was doing a manual audit - I don't recall this being caught by the IBM accessibility checker browser addon I have been using.

Unfortunately, that's not sufficient evidence for us to act on. We can't know what you're checking for, we can't automate it, and we can't use tests to not regress when behavior is being manually verified.

**Comment by cscheid** [comment]
Thanks, @glin, I can repro now.

**Comment by srvanderplas** [comment]
That makes sense, but is also tricky given that everything I've seen says
that automated tests aren't sufficient to verify accessibility, and that
manual audits are required to ensure screen reader and keyboard navigation
works as expected.

Testing for this specific thing I suspect will be easy enough to automate
-- I'm just hitting the tab button over and over and seeing if focus
changes include selecting the tab elements, so that is absolutely testable
with something like webdriver.

I fully agree that the recommendations are entirely counter to the
philosophy of continuous integration. I was honestly surprised to see that
the automatic checker I was using didn't flag this one at all. And, since I
need to use the bootstrap theme to get sufficient contrast (or recreate it
entirely, I guess?) to pass other checks, it's important that using a
bootstrap theme doesn't completely destroy keyboard navigation
functionality.

**Comment by srvanderplas** [comment]
Is there any update that will allow bootstrap themes with tabbing, given @glin 's reprex?

**Comment by mcanouil** [comment]
@srvanderplas 

We appreciate your interest in the project, but I'd like to share some guidance on how we can make discussions more productive.

Posts asking "Any news?" or "When will this be ready?" without specific questions or constructive feedback don't help move development forward.
They create notification noise for maintainers without providing actionable input that could influence development priorities.

Rather than asking for general updates, consider asking specific technical questions about implementation, contributing implementation suggestions, or submitting pull requests for features you need.

The most valuable contributions focus on "here's how I can help solve this" rather than "here's what I want you to do for me."

For the latest information about progress and planned features, check the project's changelog, release notes, or issues.

Thanks for understanding and for your continued interest in the project.

**Comment by cscheid** [comment]
This is fixed on main (so it was a duplicate of another tabbing issue we fixed):

https://github.com/user-attachments/assets/b63a7b43-64f6-46b4-8565-197ecdff47a3
